### PR TITLE
CASMINST-4256: Add documentation for restoring NTP data in BSS

### PR DESCRIPTION
**Pull request by manderson-hpe**
## Summary and Scope

Add documentation that helps a user restore NTP data to BSS if it is missing.

## Issues and Related PRs

_List and characterize relationship to Jira/Github issues and other pull requests. Be sure to list dependencies._

* Resolves [CASMINST-4256](https://jira-pro.its.hpecorp.net:8443/browse/CASMINST-4256)

### Tested on:

  * `Rocket`

## Pull Request Checklist

- [ ] Version number(s) incremented, if applicable
- [ ] Copyrights updated
- [ ] License file intact
- [ ] Target branch correct
- [ ] CHANGELOG.md updated
- [ ] Testing is appropriate and complete, if applicable
- [ ] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

**Review by jacobsalmela — APPROVED** [review]

File: operations/node_management/Configure_NTP_on_NCNs.md [hunk 1]
```diff
@@ -198,6 +198,40 @@ peer ncn-w003 minpoll -2 maxpoll 9 iburst
 
 ##### Quick Fixes
 
+###### Fix BSS Metadata
+
+If nodes are missing metadata for NTP, you will be required to generate the data using `csi` and your system's `system_config.yaml`. If you do not have your seed data in the `system_config.yaml` then you will need to open a ticket to help generate the NTP data.
+
+1. Update system_config.yaml to have the correct NTP settings
+```
```
[the remark this answers is not in the document]
Comment (manderson-hpe) [reply operations/node_management/Configure_NTP_on_NCNs.md:206]: Why does the code block need to be indented? Isn't the indented code block for sections of code that are not in the 3 backticks? I notice that there are varying indentations for code blocks in the documents.

File: operations/node_management/Configure_NTP_on_NCNs.md [hunk 2]
```diff
@@ -198,6 +198,40 @@ peer ncn-w003 minpoll -2 maxpoll 9 iburst
...
+ntp-servers:
+  - ncn-m001
+  - ntp.hpecorp.net
+ntp-timezone: UTC
+```
+2. Run `csi config init` to generate new configs
```
[the remark this answers is not in the document]
Comment (manderson-hpe) [reply operations/node_management/Configure_NTP_on_NCNs.md:212]: Done.

File: operations/node_management/Configure_NTP_on_NCNs.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (nrockershousen) [reply operations/node_management/Configure_NTP_on_NCNs.md:206]: It should be indented when it is part of a step. That way it is easier for the reader to tell what belongs to each step.

File: operations/node_management/Configure_NTP_on_NCNs.md [hunk 3]
```diff
@@ -198,6 +198,40 @@ peer ncn-w003 minpoll -2 maxpoll 9 iburst
...
+3. In the newly created `system/basecamp` directory execute `upgrade_ntp_timezone_metadata.sh`
+4. Find the relevant file(s) to the node(s) with missing metadata, such as `upgrade-metadata-000000000000.json` based on the mac address of the node.
+5. Find the xname for the node that needs to be fixed
+`cat /etc/cray/xname` on the relevant node
+6. From `m001` execute the following command to update BSS:
```
[the remark this answers is not in the document]
Comment (manderson-hpe) [reply operations/node_management/Configure_NTP_on_NCNs.md:217]: Done.

File: operations/node_management/Configure_NTP_on_NCNs.md [hunk 4]
```diff
@@ -198,6 +198,40 @@ peer ncn-w003 minpoll -2 maxpoll 9 iburst
...
+`csi handoff bss-update-cloud-init --user-data="upgrade-metadata-000000000000.json" --limit=<xname>`
+7. Obtain the new cloud-init code and templates. If your system is air-gapped then you can obtain these files from the CSM tarball in the scripts directory.
+```
```
[the remark this answers is not in the document]
Comment (manderson-hpe) [reply operations/node_management/Configure_NTP_on_NCNs.md:220]: Done.

File: operations/node_management/Configure_NTP_on_NCNs.md [hunk 5]
```diff
@@ -198,6 +198,40 @@ peer ncn-w003 minpoll -2 maxpoll 9 iburst
...
+wget -O /usr/lib/python3.6/site-packages/cloudinit/config/cc_ntp.py https://raw.githubusercontent.com/Cray-HPE/metal-cloud-init/main/cloudinit/config/cc_ntp.py`
+wget -O /etc/cloud/templates/chrony.conf.cray.tmpl https://raw.githubusercontent.com/Cray-HPE/metal-cloud-init/main/config/cray.conf.j2`
+```
+10. Continue with the upgrade
```
[the remark this answers is not in the document]
Comment (manderson-hpe) [reply operations/node_management/Configure_NTP_on_NCNs.md:224]: This documentation is specific to upgrading nodes and the BSS data being missing. So the admin would need to stop here, continue the upgrade, and then if needed execute the remainder of the steps.

File: operations/node_management/Configure_NTP_on_NCNs.md [hunk 6]
```diff
@@ -198,6 +198,40 @@ peer ncn-w003 minpoll -2 maxpoll 9 iburst
...
+11. When the upgrade is completed there might be extra files and configurations, execute the following to clean up NTP if required.
+12. Remove `pool.conf` on all nodes:
```
[the remark this answers is not in the document]
Comment (manderson-hpe) [reply operations/node_management/Configure_NTP_on_NCNs.md:226]: Nice catch.

**Review by nrockershousen — APPROVED** [review]

**Review by alexanderkingh — APPROVED** [review]

File: operations/node_management/Configure_NTP_on_NCNs.md [hunk 7]
```diff
@@ -198,6 +198,60 @@ peer ncn-w003 minpoll -2 maxpoll 9 iburst
...
+    ```bash
+    ncn# cat /etc/cray/xname
+    ```
+6. From `ncn-m001` execute the following command to update BSS:
+    ```bash
+    ncn# csi handoff bss-update-cloud-init --user-data="upgrade-metadata-000000000000.json" --limit=<xname>`
+    ```
+7. Obtain the new cloud-init code and template files from the scripts directory in the CSM documentation tarball. Otherwise, you can download the latest versions from Github:
+    ```bash
+    ncn# wget -O /usr/lib/python3.6/site-packages/cloudinit/config/cc_ntp.py https://raw.githubusercontent.com/Cray-HPE/metal-cloud-init/main/cloudinit/config/cc_ntp.py`
```
Comment (gbaker-hpe) [inline operations/node_management/Configure_NTP_on_NCNs.md:233]: We need to not instruct admins to go out to Github and we shouldn't reference the main branch which could drastically change at any time.

**Review by gbaker-hpe — APPROVED** [review]

**Pull request merged**